src/array_api_extra/_lib/_funcs.py

```diff
@@ -29,6 +35,157 @@
 ]
 
 
+@overload
+def apply_where(  # type: ignore[no-any-explicit,no-any-decorated] # numpydoc ignore=GL08
+    cond: Array,
+    f1: Callable[..., Array],
+    f2: Callable[..., Array],
+    args: Array | tuple[Array, ...],
+    /,
+    *,
+    xp: ModuleType | None = None,
+) -> Array: ...
+
+
+@overload
+def apply_where(  # type: ignore[no-any-explicit,no-any-decorated] # numpydoc ignore=GL08
+    cond: Array,
+    f1: Callable[..., Array],
+    args: Array | tuple[Array, ...],
+    /,
+    *,
+    fill_value: Array | int | float | complex | bool,
```

Comment:
I think I'd support the overloading the third positional argument could be used for either `fill_value` or `f2` because they are conceptually similar. But I am getting confused in gh-22557 by the third positional argument being `args` sometimes.

Reply:
I think that would be problematic. `apply_where(cond, func, x, y)` looks extremely obscure to me. It's calling neither `f(x, y)` nor `f(x) if cond else y` as one would intuitively read it; it's calling `f(y) if cond else x`.
By contrast, `apply_where(cond, func, y, fill_value=x)` looks unambiguous, also to a distracted reader.